public/js/review-searchBar.js

```diff
@@ -0,0 +1,125 @@
+import { CLASS_LIST } from './util/cssClasses.js';
+import { delay } from './util/util.js';
+import { URL, KEYCODE } from './util/data.js';
+
+class SearchBar {
+  constructor({ domElem, inputArea, suggestions, rollingKeywords }) {
+    this.domElem = domElem;
+    this.inputArea = inputArea;
+    this.suggestions = suggestions;
+    this.rollingKeywords = rollingKeywords;
+  }
+
+  registerEvent() {
+    const { HIDDEN } = CLASS_LIST;
+
+    this.inputArea.addEventListener('focus', () => {
+      this.suggestions.classList.remove(HIDDEN);
+      this.rollingKeywords.classList.add(HIDDEN);
+    })
+
+    this.domElem.addEventListener('mouseleave', () => {
+      if(this.inputArea.value === '') {
+        this.inputArea.blur();
+        this.suggestions.classList.add(HIDDEN);
+        this.rollingKeywords.classList.remove(HIDDEN);
+      }
+      if(this.inputArea.value) {
+        this.inputArea.blur();
+        this.suggestions.classList.add(HIDDEN);
+      }
+    });
+  }
+}
+
+class Rolling {
+  constructor({ domElem, rollingData }) {
+    this.domElem = domElem;
+    this.rollingData = rollingData;
+    this.currItemIdx = 0;
+    this.currY = 4;
+    this.heightOfOneItem = 30;
+    this.init();
+  }
+
+  render() {
+    let renderingValue = this.rollingData.reduce((prev, curr, idx) => {
+      return prev + `<li><span class="num-rank">${idx + 1}</span>${this.rollingData[idx]}</li>`
+    }, '')
+    renderingValue += `<li><span class="num-rank">1</span>${this.rollingData[0]}</li>`
+    this.domElem.innerHTML = renderingValue;
+  }
+
+  async roll() {
+    await delay('', 2000);
+    const rollngTimeoutId = setTimeout(() => {
+      this.setAnimation();
+      this.currItemIdx++;
+      if(this.currItemIdx === 11) {
+        this.resetStates();
+      }
```

Comment:
10 이라는 값도 상수로 선언해서 매직넘버를 최대한 줄이면 좋을 것 같아요.
```suggestion
      if(this.currItemIdx === MAX_ITEM_COUNT) {
        this.resetStates();
      } else {
        this.currItemIdx++;
      }
      
```

Reply:
오오! 얼른 고쳐야겠습니다!! 갑자기 궁금한 게 생겼는데 `MAX_ITEM_COUNT` 상수 선언은 따로 `constant.js`와 같은 모듈에서 선언해두는 게 좋을지, 아니면 클래스가 존재하는 모듈에 모듈전역변수(?)처럼 해주는 게 좋을지, 클래스의 constructor안에 넣어서 property로 설정하는 게 좋을지 궁금합니다 ㅎㅎㅎ

Reply:
어플리케이션 전체적으로 공유되는 값이면 constant.js와 같은 파일에 정의하는게 좋을 것 같아요. `MAX_ITEM_COUNT`는 `Rolling`의 관심사이지만 각각의 인스턴스마다 가지고 있는 상태가 아니라 각각의 클래스가 공유해야하는 값이니 [static public fields](https://github.com/tc39/proposal-static-class-features/#static-public-fields)가 어울릴 것 같네요.

Reply:
static이 뭔지 항상 그냥 넘겼었는데 이번 기회에 확실히 알고 갑니닷!!! 감사합니다 🥺🥺🥺